posts/archive/ansible-tips-and-tricks-dealing-with-unreliable-connections-and-services.md

```diff
@@ -0,0 +1,260 @@
+      stdout: ''
+      stdout_lines: 
+    ...ignoring
+
+    TASK [debug task_register_var] ****************************************************
+      msg:
+        attempts: 5
+        changed: true
+```
+
+In the Ansible workshops I am actually using this combination of error
+handling for Let's Encrypt to make it easy for Ansible users to
+troubleshoot the issue.  If there are any tasks that have a failure that
+can be skipped, I can add it to a variable and print it at the end of
+the workshop playbook (the playbook responsible for provisioning
+instances for students to use).
+
+```yml
+    - name: failure test playbook
+      hosts: localhost
+      gather_facts: false
+      vars:
+        summary_information: |
+          PROVISIONER SUMMARY
+          *******************
+
+      tasks:
+        - name: ISSUE CERT
+          shell: certbot certonly --standalone -d student1.testworkshop.rhdemo.io --email ansible-network@redhat.com --noninteractive --agree-tos
+          register: issue_cert
+          until: issue_cert is not failed
+          retries: 5
+          ignore_errors: yes
+
+        - name: set facts for output
+          set_fact:
+          summary_information: |
+            {{summary_information}}
+            - The Lets Encrypt certbot failed, please check https://letsencrypt.status.io/ to make sure the service is running
+          when: issue_cert is failed
+
+        - name: print out summary information
+          debug:
+            msg: "{{summary_information}}"
+```
+
+This prints out a very easy to understand message to the terminal window:
+
+![Terminal
+Readout](https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=884&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png){style="width: 884px;"
+width="884"
+srcset="https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=442&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png 442w, https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=884&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png 884w, https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=1326&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png 1326w, https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=1768&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png 1768w, https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=2210&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png 2210w, https://www.ansible.com/hs-fs/hubfs/Images/blog-social/Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png?width=2652&name=Ansible-Networking-Tips-Tricks-Terminal-Screenshot.png 2652w"
```

Comment:
hubspot specific links, display broken. 
we have to search for `/hs-fs/` and `hubfs/`, as those link to hubspot assets.